patches/tModLoader/Terraria/ModLoader/UI/ModBrowser/UIModBrowser.cs

```diff
@@ -184,103 +236,139 @@ public void BackClick(UIMouseEvent evt, UIElement listeningElement)
 
 	private void ReloadList(UIMouseEvent evt, UIElement listeningElement)
 	{
-		if (Loading) return;
-		SoundEngine.PlaySound(SoundID.MenuOpen);
-		PopulateModBrowser(uiOnly: false);
+		if (Loading) {
```

Comment:
See related comment about button function being tied to display state.

Reply:
Now (one of the last commits) that `UIAsyncList.State` is no longer the task internal state but the graphic state updated in Update correctly the state matches the UI in any case (if I misunderstood the comment let me know)